packages/api/tests/lib/writeRecords/test-write-granules.js

```diff
@@ -1064,3 +1125,119 @@ test.serial('updateGranuleStatusToQueued() throws error if record does not exist
     }
   );
 });
+
+test.serial('updateGranuleStatusToFailed() updates granule status in the database', async (t) => {
+  const {
+    knex,
+    collectionCumulusId,
+    granule,
+    granuleId,
+    granuleModel,
+    granulePgModel,
+  } = t.context;
+  await writeGranuleFromApi({ ...granule }, knex);
+  const dynamoRecord = await granuleModel.get({ granuleId });
+  const postgresRecord = await granulePgModel.get(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  );
+
+  await updateGranuleStatusToFailed({ granule: dynamoRecord, knex });
+  const updatedDynamoRecord = await granuleModel.get({ granuleId });
+  const updatedPostgresRecord = await granulePgModel.get(
+    knex,
+    { granule_id: granuleId, collection_cumulus_id: collectionCumulusId }
+  );
+  const omitList = ['execution', 'status', 'updatedAt', 'updated_at'];
+  t.falsy(updatedDynamoRecord.execution);
+  t.is(updatedDynamoRecord.status, 'failed');
+  t.is(updatedPostgresRecord.status, 'failed');
+  t.deepEqual(omit(dynamoRecord, omitList), omit(updatedDynamoRecord, omitList));
+  t.deepEqual(omit(postgresRecord, omitList), omit(updatedPostgresRecord, omitList));
+});
+
+test.serial('updateGranuleStatusToFailed() throws error if record does not exist in pg', async (t) => {
+  const {
+    knex,
+    granule,
+    granuleId,
+  } = t.context;
+
+  await writeGranuleFromApi({ ...granule }, knex);
+
+  const name = randomId('name');
+  const version = randomId('version');
+  const badGranule = fakeGranuleFactoryV2({
+    granuleId,
+    collectionId: constructCollectionId(name, version),
+  });
+  await t.throwsAsync(
+    updateGranuleStatusToFailed({ granule: badGranule, knex }),
+    {
+      name: 'RecordDoesNotExist',
+      message: `Record in collections with identifiers {"name":"${name}","version":"${version}"} does not exist.`,
+    }
+  );
+});
+
+test.serial('updateGranuleStatus() updates granule status in the database', async (t) => {
+  const {
+    knex,
+    collectionCumulusId,
+    granule,
+    granuleId,
+    granuleModel,
+    granulePgModel,
+  } = t.context;
+  await writeGranuleFromApi({ ...granule }, knex);
+  const dynamoRecord = await granuleModel.get({ granuleId });
+  const postgresRecord = await granulePgModel.get(
```

Comment:
We should verify that the status of these records is not `completed` so we can be sure that `updateGranuleStatus` is what is doing the updates

Reply:
great catch. thanks.